File: R/8kyu/8kyu-convert-boolean-values-to-strings-yes-or-no.r
# Convert boolean values to strings 'Yes' or 'No'
# Difficulty: 8 kyu

# Solution 1
bool_to_word <- function(bool){
  if (bool == T){
    return("Yes")
  }else{
    return("No")
  }
}

# Solution 2 (more concise)
bool_to_word <- function(bool){
  ifelse(bool, "Yes", "No")
}


# Test

test_that('Basic tests', {
  expect_equal(bool_to_word(TRUE), 'Yes')
  expect_equal(bool_to_word(FALSE), 'No')
})

File: Julia/8kyu/8kyu-convert-boolean-values-to-strings-yes-or-no.jl
# Convert boolean values to strings 'Yes' or 'No'
# Language: Julia
# Version: 1.5
# Difficulty: 8 kyu

module BoolWord
    export bool_to_word 
    function bool_to_word(boolean::Bool)::String
      ifelse(boolean::Bool, "Yes", "No")
    end
end


# Tests
using FactCheck
using .BoolWord


facts("Testing Boolean to Word") do
  @fact bool_to_word(true) --> "Yes"
  @fact bool_to_word(false) --> "No"
end